plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotProjectActions.java

```diff
@@ -197,27 +200,36 @@ public static SWTBotTree getProjectRootTree(SWTWorkbenchBot bot) {
     return new SWTBotTree(tree);
   }
 
-  /**
-   * Returns true if there are errors in the Problems view. Returns false otherwise.
-   */
-  public static List<String> getErrorsInProblemsView(SWTWorkbenchBot bot) {
-    // Open Problems View by Window -> show view -> Problems
-    bot.menu("Window").menu("Show View").menu("Problems").click();
-
-    SWTBotView view = bot.viewByPartName("Problems");
-    view.show();
-    SWTBotTree tree = view.bot().tree();
-
-    List<String> errors = new ArrayList<>();
-    
-    for (SWTBotTreeItem item : tree.getAllItems()) {
-      String text = item.getText();
-      if (text != null && text.startsWith("Errors")) {
-        errors.add(text);
+  /** Return the list of all problems in the workspace. */
+  public static List<String> getAllBuildErrors(SWTWorkbenchBot bot) {
+    IWorkspaceRoot root = getWorkspaceRoot();
+    List<String> foundProblems = new LinkedList<>();
+    for (IProject project : root.getProjects()) {
+      try {
+        IMarker[] problems = project.findMarkers(IMarker.PROBLEM, true /* includeSubtypes */,
+            IResource.DEPTH_INFINITE);
+        for (IMarker problem : problems) {
+          int severity = problem.getAttribute(IMarker.SEVERITY, IMarker.SEVERITY_INFO);
+          if (severity >= IMarker.SEVERITY_ERROR) {
+            foundProblems.add(formatProblem(problem));
+          }
+        }
+      } catch (CoreException ex) {
```

Comment:
It's a checked exception, so it would require percolating that up.  I can't think of a reason why or how we would create erroring projects, so it didn't seem worth exposing it.